### ISSUE TITLE
Rate Limit Error

### ISSUE THREAD

**Issue by emremutlu08** [RooVetGit/Roo-Cline#196]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5 Sonnet

### What happened?

![Screenshot 2024-12-22 at 14 30 18](https://github.com/user-attachments/assets/6ac83560-f6a2-4b9d-869f-c68e5e797853)

Issue Description:
When using the Roo-Cline package, I encountered the following error after reaching the rate limit:

`429 {"type":"error","error":{"type":"rate_limit_error","message":"This request would exceed your organization’s rate limit of 80,000 input tokens per minute. For details, refer to: https://docs.anthropic.com/en/api/rate-limits; see the response headers for current usage. Please reduce the prompt length or the maximum tokens requested, or try again later. You may also contact sales at https://www.anthropic.com/contact-sales to discuss your options for a rate limit increase."}}`
The error itself is expected, but the current behavior requires manually clicking the "Retry" button every time.

Desired Behavior:
I would like to see an automated retry mechanism that waits for 1 minute after receiving this error and then retries the request without user intervention.

### Steps to reproduce

1. 
2.
3.

**Comment by mrubens** [comment]
Good idea. This seems similar to https://github.com/RooVetGit/Roo-Cline/pull/116... I just don't love the code and haven't really tested it. If you're experiencing this frequently would you want to give a custom VSIX with this change a shot to see if it helps?

**Comment by emremutlu08** [comment]
Sure, I can try it on weekend.

**Comment by mrubens** [comment]
I'm going to close this now that the feature is available - please reopen if it's not meeting your needs. Thank you!
